routers/web/user/setting/keys.go

```diff
@@ -78,6 +79,11 @@ func KeysPost(ctx *context.Context) {
 		ctx.Flash.Success(ctx.Tr("settings.add_principal_success", form.Content))
 		ctx.Redirect(setting.AppSubURL + "/user/settings/keys")
 	case "gpg":
+		if !setting.User.Enabled("gpg keys") {
```

Comment:
It may be better to use constants for these names (can be used in the router too).

Reply:
Done.